example/objc/main_darwin.go

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2022 The Ebitengine Authors
+
+package main
+
+import (
+	"fmt"
+	"unsafe"
+
+	"github.com/ebitengine/purego/objc"
+)
+
+type barObject struct {
+	isa objc.Class `objc:"BarObject:NSObject <NSDelegateWindow>"`
+	bar int
+}
+
+func (b *barObject) Init(_cmd objc.SEL) objc.ID {
+	return objc.ID(unsafe.Pointer(b)).SendSuper(_cmd)
+}
+
+func (b *barObject) Bar(_cmd objc.SEL) int {
+	return b.bar
+}
+
+func (b *barObject) SetBar(_cmd objc.SEL, bar int) {
+	b.bar = bar
+}
+
+func (_ *barObject) Selector(metName string) objc.SEL {
```

Comment:
Is `Selector` called once for one name? i.e. isn't `RegisterName` called twice or more for the same method?

Reply:
What if `Selector` returns a string instead of `objc.SEL`?

Reply:
If it is enough to return a string, would it be possible to calculate a method name automatically? I feel like

1. Lower the first character
2. add `:` and argument names if necessary

sounds enough. If a method takes two ore more arguments, things would not be so easy though.

Reply:
Yeah my original idea was to add : after if it took arguments. I sadly couldn't think of a good way to handle determining the name for multiple returns tho. Returning a SEL allows the user to optimize bc they may have a global variable with the selector and so RegisterName would only be called once. It also allows for a potential where the SEL is determined in a way we haven't even thought of yet. But if we return just a string then we guarantee to have to call RegisterName

Reply:
>  I sadly couldn't think of a good way to handle determining the name for multiple returns tho.

I was talking about mulitiple arguments, not returning values.

Reply:
Yep! I was thinking arguments but accidentally wrote return. ObjC doesn't have multiple returns. 

I see no easy way to determine the ObjC name based on the argument list and name of the Go method

Reply:
OK.

> It also allows for a potential where the SEL is determined in a way we haven't even thought of yet. But if we return just a string then we guarantee to have to call RegisterName

I slightly prefer returning a string to objc.SEL. The risk is, as you said, we would not be able to have a flexible way to get SEL. What do you think?

Reply:
Why do you think returning string is more appropriate? Is it because `Selector(string) string` is a possible Interface you'd see in the wild?

Reply:
It's because most of the implementation would just return SELs generated by `objc.RegisterName` so returning SELs seemed redundant.

Reply:
I foresee most return a variable that was created using RegisterName so this way each method's RegisterName is only called once instead of twice

Reply:
My point is

```go
	switch metName {
	case "Init":
		return "init"
	case "SetBar":
		return "setBar:"
	case "Bar":
		return "bar"
	default:
		return ""
	}
```

would be simpler and easier for those who have to implement this than

```go
	switch metName {
	case "Init":
		return objc.RegisterName("init")
	case "SetBar":
		return objc.RegisterName("setBar:")
	case "Bar":
		return objc.RegisterName("bar")
	default:
		return 0
	}
```

Reply:
But, I am not confident this is a good idea. For example, returning SEL might be more correct in terms of semantics.

Reply:
True if that is the only case of using the Selector. However if you want to call the method that matches that selector you'll likely need a global variable. Global variables are currently used in [ebitengine](https://github.com/hajimehoshi/ebiten/blob/50021ef3e4e3028e81ee77e3763a2b248cc8666f/internal/cocoa/api_cocoa_darwin.go#L36).

```go
var sel_init  = objc.RegisterName("init")
// others...

        switch metName {
	case "Init":
		return sel_init
	case "SetBar":
		return sel_setBar
	case "Bar":
		return sel_bar
	default:
		return 0
	}
```

Reply:
Yeah, it'd be nice if the example reflects this.

But, we still need `RegisterName` for global variables, right? Or do we use the selectors for other usecaes?

Reply:
RegisterName needs to be called somewhere bc it's not possible to call an ObjC method without one. 

I don't think there really are any other uses for selectors other than calling ObjC methods